### ISSUE TITLE
Linear decay instead of cosine

### ISSUE THREAD

**Issue by dirkgr** [allenai/OLMo#310]

**Comment by ayushbits** [comment]
Hi @epwalsh 
Contrary to most of the models, OLMO followed linear LR decay instead of cosine. It would be interesting to observe the ablations. Can you post them if ablations are ready or post your observations here ?

**Comment by epwalsh** [comment]
@ayushbits we'll probably write about this in our upcoming "training details" paper. In the meantime I can give you my opinion, but take this with a grain a salt since our ablation at the 7B scale is not quite finished:

- We used a linear decay at the time for our 7B because we didn't know which would work better, and it was a little safer to use linear instead of cosine since cosine initially keeps the learning rate higher for longer.
- Since then we've discovered other reasons why some of our early experiments had spikes, and we're now a lot more confident in our ability to find stable hyperparameters. If we knew then what we do now we'd have probably used cosine.
- But so far it seems like it really doesn't matter that much, and AFAIK there's no theoretical justification for either (though there is a theoretical justification for an inverse square root schedule - Will Merrill has a paper about this - but that's not as popular in practice).

**Comment by ayush-vizzhy** [comment]
Thanks Pete @epwalsh for the detailed explanation. Helps a lot !

**Comment by ayushbits** [comment]
Thanks Pete @epwalsh  for the detailed explanation. Helps a lot !

**Comment by dumitrac** [comment]
Marking the items prior to Feb 29th as "closed".
